wdl-lint/src/rules/runtime_section_keys.rs

```diff
@@ -0,0 +1,355 @@
+    }
+}
+
+/// A utility method to parse the recommended keys from a static set of runtime
+/// keys from either WDL v1.0 or WDL v1.1.
+fn recommended_keys<T: Deref<Target = HashSet<Key>>>(keys: T) -> HashSet<Key> {
+    keys.clone()
+        .into_iter()
+        .filter(|key| key.is_recommended())
+        .collect::<HashSet<Key>>()
+}
+
+impl Visitor for RuntimeSectionKeysRule {
+    type State = Diagnostics;
+
+    fn document(
+        &mut self,
+        _: &mut Self::State,
+        reason: wdl_ast::VisitReason,
+        document: &wdl_ast::Document,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        // Reset the visitor upon document entry.
+        *self = Default::default();
+
+        // NOTE: this rule is dependent on the version of the WDL document.
+        self.version = document
+            .version_statement()
+            .and_then(|s| s.version().as_str().parse().ok());
+    }
+
+    fn task_definition(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        _: &TaskDefinition,
+    ) {
+        match reason {
+            VisitReason::Enter => {
+                self.current_runtime_span = None;
+                self.keys_seen = Default::default();
+            }
+            VisitReason::Exit => {
+                // If a runtime section span has not been encountered, then
+                // presumably there won't be any place to put missing
+                // recommended runtime keys.
+                let runtime_span = match self.current_runtime_span {
+                    Some(span) => span,
+                    None => return,
+                };
+
+                if let Some(SupportedVersion::V1(minor_version)) = self.version {
+                    let recommended_keys = match minor_version {
+                        V1::Zero => recommended_keys(&V1_0_KEYS.clone()),
+                        V1::One => recommended_keys(&V1_1_KEYS.clone()),
+                        _ => Default::default(),
+                    };
+
+                    for key in recommended_keys {
+                        if !self.keys_seen.contains_key(key.name()) {
+                            state.add(missing_recommended_key(
+                                key.name(),
+                                runtime_span,
+                                &format!("the {minor_version} specification"),
+                            ));
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn runtime_section(
+        &mut self,
+        _: &mut Self::State,
+        reason: VisitReason,
+        section: &RuntimeSection,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        match self.current_runtime_span {
+            // SAFETY: if this occurs, somehow two runtime sections were allowed
+            // to be parsed successfully within a WDL document. This should be
```

Comment:
Any number of runtime sections are permissible in a WDL document. Only one is permitted _per task_. I'm not sure why this `unreachable!()` isn't triggering in Arena 🤔

Reply:
Ah—it's because it _was_ triggering them, but then I realized this mistake and changed it to be per-`task`. I just forgot to update this comment, thank you.